server/src/main/java/org/opensearch/cluster/metadata/IndexTemplateMetadata.java

```diff
@@ -257,6 +258,16 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeOptionalVInt(version);
     }
 
+    public void writeVerifiableTo(StreamOutput out) throws IOException {
```

Comment:
Shouldn't this be `BuffereredChecksumStreamOutput`?

Reply:
`BuffereredChecksumStreamOutput` is a part of core in 2.x, where as `VerifiableWriteable` is added in `libs`. This would be reverse dependency. Logic should still work as expected since we are passing `BuffereredChecksumStreamOutput` only from callers.

Reply:
Removed `VerifiableWriteable` interface for now and using `BuffereredChecksumStreamOutput` for `writeVerifiableTo` method arg since there is no dependency now.

Reply:
@himshikha @Bukhtawar Having `BuffereredChecksumStreamOutput` in a different location across main and 2.x will lead to automated backport failures as the code diverges. I just encountered a backport failure in #16483 due to this change. I'd encourage finding a way to do this that keeps the branches in sync, at least until we're ready to do a major version bump. Also it does seem a bit dangerous (if I understand this correctly) that we're losing the type safety offered by the `BuffereredChecksumStreamOutput` type.